### PR TITLE
feat: enum choices for list command argument (issue #732)

### PR DESCRIPTION
**Pull request by jimlloyd**
**Please describe the purpose of this pull request.**
Adds a new simple feature to make CLI help and usage nicer.

**How to test**
This is a change to the CLI only, and currently only for the `memgpt list` command. If you have any automated tests that would cover this please direct me to them.

**Have you tested this PR?**
Yes. I only tested that the CLI for `memgpt list ...` behaves correctly for the various positive and negative scenarios, i.e. all of the previously supported `memgpt list <options>` still work as expected, and when help is requested or when bad options are given an appropriate usage message is given.

**Related issues or PRs**
#732 

**Is your PR over 500 lines of code?**
No

**Additional context**
This is my first PR. I'm happy to incorporate any feedback. I'm also happy to extend this PR to cover other CLI cases, such as the `memgpt add` command.

### CODE REVIEW

File: memgpt/cli/cli_config.py
Comment (cpacker) [inline memgpt/cli/cli_config.py:489]: Do you have a screenshot for what this looks like in-context?

Wondering if this should be color coded to match other `cli.py` / `configure` elements.

Comment (jimlloyd) [reply memgpt/cli/cli_config.py:489]: Whoops that typer.echo statement was put there while debugging. I think we should remove it. But I will look into coloring and I will provide a screenshot later this morning.

**Comment by jimlloyd** [comment]
Here are screenshots in both light and dark modes. I think the default coloring is fine, but LMK if you think otherwise.

image

image

**Comment by jimlloyd** [comment]
I pushed a `fixup` commit to remove the `typer.echo()` statement. The team I work with regularly uses the practice of pushing fixup commits for code in review so that the reviewer doesn't have to re-review commits already looked at. We use the shorthand `LGTM` when a PR is approved, and `LGTM rebase` when the code is approved but fixups should be squashed before approval. LMK if you have other conventions.

**Comment by jimlloyd** [comment]
Oh, and we use the shorthand `PTAL` for Please Take a Look, which means that the author thinks all of the reviewers feedback has been addressed and is ready for another review pass.

PTAL

**Review by cpacker — APPROVED** [review]
LGTM

**Comment by cpacker** [comment]
> I pushed a `fixup` commit to remove the `typer.echo()` statement. The team I work with regularly uses the practice of pushing fixup commits for code in review so that the reviewer doesn't have to re-review commits already looked at. We use the shorthand `LGTM` when a PR is approved, and `LGTM rebase` when the code is approved but fixups should be squashed before approval. LMK if you have other conventions.

We've been using LGMT on and off. The fix-up commit tagging idea is cool. We've been squashing all PRs and generally merging even long-running feature branches instead of rebasing since it makes it easier when multiple people are actively working on the branch.

**Comment by cpacker** [comment]
Thanks for the PR!

**Pull request merged**
